### PR TITLE
vm/networks (search method)

### PR DESCRIPTION
**Pull request by harshshah-tenable**
Changes for Search method

### CODE REVIEW

File: tests/io/v3/vm/networks/test_networks_api.py [hunk 1]
```diff
@@ -207,12 +210,64 @@ def test_list_scanners(api):
...
+        match=[matchers.json_params_matcher(payload)],
+        json=api_response
+    )
+
+    iterator = api.v3.vm.networks.search(
+        fields=fields, limit=200
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    event_list = []
```
Comment (ngaglani-tenable) [inline tests/io/v3/vm/networks/test_networks_api.py:257]:
```suggestion
    network_list = []
    for network in iterator:
        network_list.append(event)
    assert len(network_list) == api_response['pagination']['total']
```

File: tests/io/v3/vm/networks/test_networks_api.py [hunk 2]
```diff
@@ -207,12 +210,64 @@ def test_list_scanners(api):
...
+    for event in iterator:
+        event_list.append(event)
+    assert len(event_list) == api_response['pagination']['total']
```
Comment (ngaglani-tenable) [inline tests/io/v3/vm/networks/test_networks_api.py]:
```suggestion
    network_list = []
    for network in iterator:
        network_list.append(event)
    assert len(network_list) == api_response['pagination']['total']
```

**Review by ngaglani-tenable — APPROVED** [review]

File: tenable/io/v3/vm/networks/api.py [hunk 3]
```diff
@@ -226,72 +230,105 @@ def unassigned_scanners(self, network_id: UUID) -> List:
...
 
-        Examples:
-            Getting the listing of all agents:
+            limit (int, optional):
+                Number of objects to be returned in each request.
+                Default and max_limit is 200.
+            next (str, optional):
+                The pagination token to use when requesting the next page of
+                results. This token is presented in the previous response.
+            return_resp (bool, optional):
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/networks/api.py:297]: Update docstring.

File: tenable/io/v3/vm/networks/api.py [hunk 4]
```diff
@@ -226,72 +230,105 @@ def unassigned_scanners(self, network_id: UUID) -> List:
...
 
-            >>> for nw in tio.v3.vm.networks.list():
-            ...     pprint(nw)
+        Returns:
+            Iterable:
+                The iterable that handles the pagination for the job.
 
-            Retrieving all of the windows agents:
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (Mangusingh-te) [inline tenable/io/v3/vm/networks/api.py]: Update docstring.

File: tests/io/v3/vm/networks/test_networks_api.py [hunk 5]
```diff
@@ -207,12 +210,64 @@ def test_list_scanners(api):
...
+
+    iterator = api.v3.vm.networks.search(
+        fields=fields, limit=200
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    networks_list = []
+    for networks in iterator:
+        networks_list.append(networks)
+    assert len(networks_list) == api_response['pagination']['total']
```
Comment (Mangusingh-te) [inline tests/io/v3/vm/networks/test_networks_api.py]: Rewrite it to get rid of loop.

**Review by Mangusingh-te — CHANGES_REQUESTED** [review]

File: tests/io/v3/vm/networks/test_networks_api.py (hunk 5 above)
Comment (harshshah-tenable) [reply tests/io/v3/vm/networks/test_networks_api.py]: Updated

File: tenable/io/v3/vm/networks/api.py (hunk 4 above)
Comment (harshshah-tenable) [reply tenable/io/v3/vm/networks/api.py]: Updated

File: tenable/io/v3/vm/networks/api.py (hunk 3 above)
Comment (harshshah-tenable) [reply tenable/io/v3/vm/networks/api.py:297]: Updated

**Pull request merged**